src/roomHandlers/fishingRoomHandler.ts

```diff
@@ -0,0 +1,128 @@
+import { RoomHandler } from './roomHandler';
+
+import {
+  Command,
+  CommandHandlersByCommandByPlayerStates,
+  DEFAULT_COMMAND,
+  NarrowedRoomState,
+  NullableCommandHandler,
+  StateDescriptionAccessor,
+} from '../utils/types';
+
+const roomType = 'fishingRoom';
+type TRoomType = typeof roomType;
+type TRoomState = NarrowedRoomState<TRoomType>;
+type TStateDescriptionAccessor = StateDescriptionAccessor<TRoomState>;
+type TCommandHandlersByCommandByPlayerStates =
+  CommandHandlersByCommandByPlayerStates<TRoomType>;
+type TNullableCommandHandler = NullableCommandHandler<TRoomType>;
+
+const stateDescriptionAccessor: TStateDescriptionAccessor = {
+  AtEntrance: ({ fishCaught, brokenRod }) => {
+    const poleText = brokenRod === true ? 'broken' : 'working';
+
+    return {
+      playerStateDescription: `You are on a dock with a ${poleText} fishing pole. Total number of fish caught is ${fishCaught} so far!`,
```

Comment:
On top of this suggestion, I suggest dropping the exclamation point when `fishCaught` is 0.

```suggestion
      playerStateDescription: `You are on a dock with a ${poleText} fishing pole. You have caught ${fishCaught} fish so far!`,
```

Reply:
updated